src/configure/configure.ts

```diff
@@ -501,22 +501,24 @@ class PipelineConfigurer {
         }
 
         try {
-            let commitOrDiscard = await vscode.window.showInformationMessage(utils.format(Messages.modifyAndCommitFile, Messages.commitAndPush, this.inputs.sourceRepository.branch, this.inputs.sourceRepository.remoteName), Messages.commitAndPush, Messages.discardPipeline);
-            if (commitOrDiscard && commitOrDiscard.toLowerCase() === Messages.commitAndPush.toLowerCase()) {
-                await vscode.window.withProgress({ location: vscode.ProgressLocation.Notification, title: Messages.configuringPipelineAndDeployment }, async (progress) => {
-                    try {
-                        // handle when the branch is not upto date with remote branch and push fails
-                        this.inputs.sourceRepository.commitId = await this.localGitRepoHelper.commitAndPushPipelineFile(this.inputs.pipelineParameters.pipelineFilePath, this.inputs.sourceRepository);
-                    }
-                    catch (error) {
-                        telemetryHelper.logError(Layer, TracePoints.CheckInPipelineFailure, error);
-                        throw (error);
-                    }
-                });
-            }
-            else {
-                telemetryHelper.setTelemetry(TelemetryKeys.PipelineDiscarded, 'true');
-                throw new UserCancelledError(Messages.operationCancelled);
+            while (!this.inputs.sourceRepository.commitId) {
```

Comment:
why do we need while loop here ?

Reply:
Because we do not want the notification to go away. This is to give user the chance to fix the problem and continue the journey from there itself.